.github/workflows/build.yml

```diff
@@ -37,8 +117,115 @@ jobs:
         run: |
           ve1/bin/pr-artifact --directory=./pr --pr-number=${{ github.event.number }} --api-url=${{ github.event.pull_request._links.self.href }}
+
       - uses: actions/upload-artifact@v2
         if: always()
         with:
           name: pr
           path: pr/
+
+      - name: 'Prepare PR failure comment'
+        if: ${{ steps.verify_pr.conclusion == 'failure' }}
+        run: |
+          python3 scripts/prepare_pr_comment.py failure ${{ github.repository }}
+
+      - name: 'Prepare PR success comment'
+        if: ${{ steps.verify_pr.conclusion == 'success' }}
+        run: |
+          python3 scripts/prepare_pr_comment.py success ${{ github.repository }}
+
+      - name: 'Comment on PR'
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            var fs = require('fs');
+            var issue_number = ${{ github.event.number }};
+            var comment = fs.readFileSync('./pr/comment', {encoding:'utf8', flag:'r'});
+            github.issues.createComment({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: Number(issue_number),
+              body: comment
+            });
+
+      - name: 'Add label to PR'
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            var fs = require('fs');
+            var issue_number = ${{ github.event.number }};
+            var vendor_label = fs.readFileSync('./pr/vendor');
+            var chart_name = fs.readFileSync('./pr/chart');
+            if (vendor_label.toString() !== "" && chart_name.toString() !== "") {
+              github.issues.addLabels({
+                issue_number: Number(issue_number),
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                labels: ['authorized-request']
+            })};
+
+      - name: 'Approve PR'
+        id: approve_pr
+        if: ${{ steps.verify_pr.conclusion == 'success' }}
+        uses: hmarr/auto-approve-action@v2
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: 'Merge PR'
+        id: merge_pr
+        if: ${{ steps.approve_pr.conclusion == 'success' }}
+        uses: pascalgn/automerge-action@v0.13.1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          MERGE_METHOD: squash
+          MERGE_LABELS: ""
+
+      - if: ${{ steps.merge_pr.conclusion == 'failure' }}
+        run: |
+          echo "The 'PR has not been merged' a step has failed."
+          exit 1
+
+      - name: Block until there is no running workflow
+        uses: softprops/turnstyle@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Dump GitHub context
+        env:
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+        run: |
+          echo "$GITHUB_CONTEXT"
+
+      - name: Configure Git
```

Comment:
Do we need this step?

Reply:
Git commit will not work without this. Also, we need a way to identify who made the commit when looking at the git history.